pubsub/rabbitmq/metadata.yaml

```diff
@@ -0,0 +1,168 @@
+      Whether or not to use durable queues.
+    default: '"false"'
+    example: '"true", "false"'
+  - name: publisherConfirm
+    type: bool
+    description: |
+      If enabled, client waits for publisher confirmation after publishing
+      a message.
+    default: '"false"'
+    example: '"true", "false"'
+  - name: maxLen
+    type: number
+    description: |
+      The maximum number of messages of a queue and its dead letter queue
+      (if dead letter enabled). If both maxLen and maxLenBytes are set then
+      both will apply; whichever limit is hit first will be enforced.
+    example: '1000'
+  - name: maxLenBytes
+    type: number
+    description: |
+      Maximum length in bytes of a queue and its dead letter queue (if dead
+      letter enabled). If both maxLen and maxLenBytes are set then both will 
+      apply; whichever limit is hit first will be enforced.
+    example: '1048576'
+  - name: concurrency
+    type: string
+    description: |
+      Allows processing multiple messages in
+      parallel (limited by the app-max-concurrency annotation, if configured).
+      Set to single to disable parallel processing. In most situations there’s 
+      no reason to change this.
+    example: '"parallel", "single"'
+    default: '"parallel"'
+  - name: enableDeadLetter
+    type: bool
+    description: |
+      Enable forwarding messages that cannot be handled to a dead-letter
+      topic.
+    default: '"false"'
+    example: '"true", "false"'
+  - name: prefetchCount
+    type: number
+    description: |
+      Number of messages to prefetch. Consider changing this to a non-zero
+      value for production environments. The value of "0" means that
+      all available messages will be pre-fetched.
+    default: '0'
+    example: '2'
+  - name: exchangeKind
+    type: string
+    description: |
+      Exchange kind of the rabbitmq exchange.
+    default: '"fanout"'
+    example: '"fanout","topic"'
+  - name: deliveryMode
+    type: number
+    description: |
+      Persistence mode when publishing messages.
+      RabbitMQ treats "2" as persistent, all other numbers as non-persistent.
+    default: '0'
+    example: '2'
+  - name: autoAck
+    type: bool
+    description: |
+      Whether or not the queue consumer should auto-ack messages.
+    default: '"false"'
+    example: '"true", "false"'
+  - name: deletedWhenUnused
+    type: bool
+    description: |
+      Whether or not the queue should be configured to auto-delete.
+    default: '"true"'
+    example: '"true", "false"'
+  - name: requeueInFailure
+    description: |
+      Whether or not to requeue when sending a negative acknowledgement
+      in case of a failure.
+    type: bool
+    default: '"false"'
+    example: '"true", "false"'
+  - name: reconnectWaitSeconds
+    description: |
+      Reconnect wait in Seconds.
+    type: duration
+  - name: ttlInSeconds
+    description: |
+      Default Queue TTL.
+    type: duration
+  - name: protocol
```

Comment:
Should probably be moved to the "Authenticate Using RabbitMQ properties" auth profile since it doesn't make sense otherwise. And you can remove "If both the connectionString and this property..."